### PR TITLE
test: migrate fork-choice unit tests to vitest

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Consolidate the testing frameworks and migrate to vitest.

**Description**

Migrate `fork-choice` unit tests to vitest.

Closes #issue_number

- Run all tests

### CODE REVIEW

File: packages/fork-choice/test/unit/protoArray/computeDeltas.test.ts [hunk 1]
```diff
@@ -303,12 +303,9 @@ describe("computeDeltas", () => {
     // 1st validator is part of an attester slashing
     const equivocatingIndices = new Set([0]);
     let deltas = computeDeltas(indices.size, votes, balances, balances, equivocatingIndices);
-    expect(deltas[0]).to.be.equals(
-      -1 * (firstBalance + secondBalance),
-      "should disregard the 1st validator due to attester slashing"
-    );
-    expect(deltas[1]).to.be.equals(secondBalance, "should move 2nd balance from 1st root to 2nd root");
```
Comment (nflaig) [inline packages/fork-choice/test/unit/protoArray/computeDeltas.test.ts]: the assertions are not obvious in my opinion, would prefer to keep those as comments (or use toBeWithMessage)

File: packages/fork-choice/test/unit/protoArray/computeDeltas.test.ts [hunk 2]
```diff
@@ -303,12 +303,12 @@ describe("computeDeltas", () => {
     // 1st validator is part of an attester slashing
     const equivocatingIndices = new Set([0]);
     let deltas = computeDeltas(indices.size, votes, balances, balances, equivocatingIndices);
-    expect(deltas[0]).to.be.equals(
+    expect(deltas[0]).toBeWithMessage(
       -1 * (firstBalance + secondBalance),
-      "should disregard the 1st validator due to attester slashing"
+      "should move 2nd balance from 1st root to 2nd root"
```
Comment (nflaig) [inline packages/fork-choice/test/unit/protoArray/computeDeltas.test.ts]:
```suggestion
      "should disregard the 1st validator due to attester slashing"
```

File: packages/fork-choice/test/unit/protoArray/protoArray.test.ts [hunk 3]
```diff
@@ -107,10 +107,7 @@ describe("ProtoArray", () => {
     ];
 
     for (const [ancestorRoot, descendantRoot, isDescendant] of assertions) {
-      expect(fc.isDescendant(ancestorRoot, descendantRoot)).to.equal(
-        isDescendant,
-        `${descendantRoot} must be ${isDescendant ? "descendant" : "not descendant"} of ${ancestorRoot}`
-      );
+      expect(fc.isDescendant(ancestorRoot, descendantRoot)).toBe(isDescendant);
```
Comment (nflaig) [inline packages/fork-choice/test/unit/protoArray/protoArray.test.ts]: If this tests fails the err message will be hard to reason about compared to previous (custom) assertion err message

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.14.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.14.0) :tada:

**Pull request merged**